gensim/models/doc2vec.py

```diff
@@ -917,27 +917,24 @@ def infer_vector(self, doc_words, alpha=None, min_alpha=None, epochs=None, steps
         doctag_vectors, doctag_locks = self.trainables.get_doctag_trainables(doc_words, self.docvecs.vector_size)
         doctag_indexes = [0]
         work = zeros(self.trainables.layer1_size, dtype=REAL)
+        neu1 = None
         if not self.sg:
             neu1 = matutils.zeros_aligned(self.trainables.layer1_size, dtype=REAL)
 
         alpha_delta = (alpha - min_alpha) / max(epochs - 1, 1)
 
+        if self.sg:
```

Comment:
Cosmetically, these changes look good, but they break the build. Can you please make the unit tests pass?